zlib/project_include.cmake

```diff
@@ -0,0 +1,5 @@
+
+set(ZLIB_FOUND 1 CACHE BOOL "")
+set(ZLIB_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/zlib CACHE PATH "")
+get_filename_component(component_name ${CMAKE_CURRENT_LIST_DIR} NAME)
+set(ZLIB_LIBRARY idf::${component_name} CACHE STRING "" FORCE)
```

Comment:
Hello @igrr ,

with my limited understanding it looks ok to me. Maybe few questions if you don't mind.

1. Do we need to keep this in cache or is this intentional to prevent possible future override?
2. The `idf` prefix seems hardcoded in `esp-idf`, so it should be fine. But I'm wondering if this can change. Meaning is it possible that the target/alias will have different prefix than `idf`?
3. I guess that using e.g. `ZLIBConfig.cmake` and [CMAKE_FIND_PACKAGE_REDIRECTS_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_REDIRECTS_DIR.html#variable:CMAKE_FIND_PACKAGE_REDIRECTS_DIR)  or other cmake's ways how to find a package(config/module mode) is not suitable for this use case and IIUC it would be probably unnecessary complicated.

Thank you

Reply:
@fhrbata Sorry for missing your comment.

> Do we need to keep this in cache or is this intentional to prevent possible future override?

I will check this again, I guess it didn't work for me without defining the variables as CACHE due to the (otherwise limited) variable scope.

> The idf prefix seems hardcoded in esp-idf, so it should be fine. But I'm wondering if this can change. Meaning is it possible that the target/alias will have different prefix than idf?

This is a good point! I guess it would be unreasonable to change this prefix now, as many components were already written under the assumption that alias targets such as `idf::component_name` exist...

> CMAKE_FIND_PACKAGE_REDIRECTS_DIR

I didn't try this. I will need more time to check if this works.

For now I have removed Zlib changes from this PR, leaving just the fix to FreeType dependencies configuration, and the addition of the new example.

I will open another PR for FindPackage support once I check the two points mentioned above.